lms/djangoapps/lms_xblock/runtime.py

```diff
@@ -190,6 +189,15 @@ def set_tag(self, scope, key, value):
         )
 
 
+class SettingsService(object):
+    """
+    Allow XBlocks to access the server settings
+    """
+    def get(self, setting_name, default=None):
+        """ Gets setting value from django settings """
+        return getattr(settings, setting_name, default)
```

Comment:
Yes, I was thinking whitelist as well. However, I was thinking more along the lines of a whitelist that contains a bucket that any type of block can add arbitrary settings to (`XBLOCK_SETTINGS[block_type][setting_name] = value`).